packages/plugin-ext/src/main/browser/view/tree-views-main.tsx

```diff
@@ -132,12 +138,19 @@ export class TreeViewsMainImpl implements TreeViewsMain {
             if (event.length === 1) {
                 const { id, contextValue } = event[0] as TreeViewNode;
                 this.proxy.$setSelection(treeViewId, id, treeViewWidget.contextSelection);
+                this.selectionService.selection = {
```

Comment:
It is not good, we should only update global selection if the widget is focused. `onSelectionChanged` happens does not care about focus. It's already correctly done by `TreeWidget` automatically. I would suggest that you don't play around with the global selection yourself, but rather try to get extract what you need from the global selection populated by the tree widget.

Reply:
done